jetty-ee11/jetty-ee11-osgi/test-jetty-ee11-osgi/src/test/java/org/eclipse/jetty/ee11/osgi/test/TestOSGiUtil.java

```diff
@@ -66,57 +69,70 @@ public void stop(BundleContext context) throws Exception
-            xmlConfigs.append(";");
+            xmlFiles.add(etc.resolve("jetty-ssl.xml"));
+            xmlFiles.add(etc.resolve("jetty-ssl-context.xml"));
+            xmlFiles.add(etc.resolve("jetty-alpn.xml"));
+            xmlFiles.add(etc.resolve("jetty-https.xml"));
         }
-        xmlConfigs.append(new File(etc, jettySelectorFileName).toURI());
-        xmlConfigs.append(";");
-        xmlConfigs.append(new File(etc, "jetty-deploy.xml").toURI());
-        xmlConfigs.append(";");
-        xmlConfigs.append(new File(etc, "jetty-testrealm.xml").toURI());
+        xmlFiles.add(etc.resolve(jettySelectorFileName));
+        xmlFiles.add(etc.resolve("jetty-deploy.xml"));
+        xmlFiles.add(etc.resolve("jetty-testrealm.xml"));
+
+        // Sanity check that the referenced files actually exist.
+        xmlFiles.forEach(p -> assertTrue(p + " doesn't exists and/or isn't a file", Files.isRegularFile(p)));
+
+        // Convert list of XML to a list of URIs separated by ";" (not OS specific, but OSGI specific)
+        String xmlConfigLine = xmlFiles.stream()
+            .map(Path::toUri)
+            .map(URI::toASCIIString)
+            .collect(Collectors.joining(";"));
 
-        options.add(systemProperty(OSGiServerConstants.MANAGED_JETTY_XML_CONFIG_URLS).value(xmlConfigs.toString()));
+        options.add(systemProperty(OSGiServerConstants.MANAGED_JETTY_XML_CONFIG_URLS).value(xmlConfigLine));
         options.add(systemProperty("jetty.http.port").value("0"));
-        options.add(systemProperty("jetty.home").value(etc.getParentFile().getAbsolutePath()));
-        options.add(systemProperty("jetty.base").value(etc.getParentFile().getAbsolutePath()));
+        options.add(systemProperty("jetty.home").value(etc.getParent().toString()));
+        options.add(systemProperty("jetty.base").value(etc.getParent().toString()));
         return options;
     }
 
-    public static List<Option>  configureJettyHomeAndPortViaBootBundle(String jettyConnectorListenerFileName)
+    public static List<Option> configureJettyHomeAndPortViaBootBundle(String jettyConnectorListenerFileName)
     {
-        //use the default set of config files embedded in jetty boot jar
         List<Option> options = new ArrayList<>();
-        StringBuffer xmlConfigs = new StringBuffer();
-        xmlConfigs.append(JettyBootstrapActivator.DEFAULT_JETTY_ETC_FILES);
-        xmlConfigs.append(",");
-        //add in a couple of external files needed for testing
-        File etc = new File(FS.separators("src/test/config/etc"));
-        xmlConfigs.append(new File(etc, jettyConnectorListenerFileName).toURI());
-        xmlConfigs.append(",");
-        xmlConfigs.append(new File(etc, "jetty-testrealm.xml").toURI());
-
-        options.add(systemProperty(OSGiServerConstants.MANAGED_JETTY_XML_CONFIG_URLS).value(xmlConfigs.toString()));
+        List<Path> xmlFiles = new ArrayList<>();
+        Path etc = MavenPaths.projectBase().resolve("src/test/config/etc");
```

Comment:
Not correct, see previous comment.

Reply:
Ditto to above, the previous code did this, and this works with the test cases.
You might be thinking of something else.